### PR TITLE
Fix Docker configuration for DISABLE_AUTHENTICATION

### LINKED ISSUE #119: DISABLE_AUTHENTICATION no longer works
Docker version: 3598be715f32

How did you run rest-server exactly?
------------------------------------
Docker

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Filesystem

Expected behavior
-----------------
Flag referenced in README to be used to disable authentication.

Actual behavior
---------------
```
web_1  | Data directory: /data
web_1  | Authentication enabled
web_1  | error: cannot load .htpasswd (use --no-auth to disable): stat /data/.htpasswd: no such file or directory
```

Steps to reproduce the behavior
-------------------------------
```
version: '3'
services:
  web:
    image: restic/rest-server
    restart: unless-stopped
    ports:
      - "8000:8000"
    environment:
      DISABLE_AUTHENTICATION: "true"
    volumes:
      - "/dir:/data"
```

Do you have any idea what may have caused this?
-----------------------------------------------
Regression

Do you have an idea how to solve the issue?
-------------------------------------------
Not at an immediate glance.

Did rest-server help you today? Did it make you happy in any way?
-----------------------------------------------------------------
I love the project(s)!

### PR DESCRIPTION
**Pull request by lwis**
What is the purpose of this change? What does it change?
--------------------------------------------------------

Fixes #119, new configuration options broke `DISABLE_AUTHENTICATION`. This ensures backwards compatibility.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

#119

Checklist
---------

- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/rest-server/blob/master/changelog/TEMPLATE))
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/rest-server/commits/master)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: changelog/unreleased/fix-docker [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix Docker configuration for DISABLE_AUTHENTICATION
```
Comment (rawtaz) [inline changelog/unreleased/fix-docker:1]: Thanks for adding a changelog entry, this is totally needed for this change. However, please use the template file at https://github.com/restic/rest-server/blob/master/changelog/TEMPLATE (copy it into this file and edit the text), and also rename this file to "issue-119".

Comment (lwis) [reply changelog/unreleased/fix-docker]: Sorry, I think I've resolved it.

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,9 @@
 FROM alpine
 
-ENV DATA_DIRECTORY /data
-ENV PASSWORD_FILE /data/.htpasswd
+ENV DATA_DIRECTORY="/data"
+ENV PASSWORD_FILE="/data/.htpasswd"
+ENV DISABLE_AUTHENTICATION=""
```
Comment (MichaelEischer) [inline Dockerfile:5]: Is it necessary to predefine all environement variables? Or is the idea here to make the available ones more visible?

File: changelog/unreleased/issue-119 [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Security: Fix Docker configuration for DISABLE_AUTHENTICATION
```
Comment (MichaelEischer) [inline changelog/unreleased/issue-119:1]: This is a `Bugfix`, after all the bug made things too secure and not insecure.

File: changelog/unreleased/issue-119 [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+
+A regression was introduced which caused the DISABLE_AUTHENTICATION environment variable to stop working in Docker configurations, this was due to a new flag being introduced to disable authentication --no-auth.
```
Comment (MichaelEischer) [inline changelog/unreleased/issue-119:3]: Please wrap lines after approx. 80 characters and split this sentence into two shorter ones.

**Review by MichaelEischer — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (pgaskin) [reply Dockerfile:5]: Also note that it's possible to reduce the number of layers created by creating the variables in a single `ENV` directive (just remove all but the first `ENV` then add a backslash to each line excluding the last line).

Comment (MichaelEischer) [reply Dockerfile:5]: Does the number of metadata layers of a docker image matter all? AFAIK the number of layers is only relevant for the filesystem overlays?

Comment (pgaskin) [reply Dockerfile:5]: You're correct. Apparently, it's been that way since Docker 1.10 in 2016 and I never noticed.

**Review by MichaelEischer — APPROVED** [review]
LGTM. I've rebased and squashed the original commits, and fixed the review comments myself.

**Pull request merged**